power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/shared_solver_functions.hpp

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_SHARED_SOLVER_FUNCTIONS_HPP
+#define POWER_GRID_MODEL_SHARED_SOLVER_FUNCTIONS_HPP
+
+#include "y_bus.hpp"
+
+#include "../calculation_parameters.hpp"
+
+namespace power_grid_model::shared_solver_functions {
```

Comment:
maybe rename this to 

```suggestion
namespace power_grid_model::common_solver_functions {
```